src/hooks.ts

```diff
@@ -177,24 +177,25 @@ export function useMMKVObject<T>(
   key: string,
   instance?: MMKV
 ): [value: T | undefined, setValue: (value: T | undefined) => void] {
-  const [string, setString] = useMMKVString(key, instance);
+  const [json, setJson] = useMMKVString(key, instance);
+
+  const value = useMemo(() => (json ? JSON.parse(json) : undefined), [json]);
 
-  const value = useMemo(() => {
-    if (string == null) return undefined;
-    return JSON.parse(string) as T;
-  }, [string]);
   const setValue = useCallback(
-    (v: T | undefined) => {
-      if (v == null) {
-        // Clear the Value
-        setString(undefined);
+    (v: (T | undefined) | ((prev: T | undefined) => T | undefined)) => {
+      if (v instanceof Function) {
```

Comment:
```suggestion
      if (typeof v === 'function') {
```

Let's use the `typeof` check instead, we used that everywhere else as well. Theoretically C++ host functions might not be function instances, and I think that check is safer.